apps/dev-portal-web/domains/docs/components/widgets/AmountDistributionCalculator/DistributionItem.tsx

```diff
@@ -0,0 +1,110 @@
+    const [bankAccount] = useState<string>(item.recipient.bankAccount)
+    const [amount, setAmount] = useState<string>(item.amount)
+    const [isFeePayer, setIsFeePayer] = useState<boolean | undefined>(item.isFeePayer)
+    const [order, setOrder] = useState<number | undefined>(item.order || 0)
+    const [vor, setVor] = useState<boolean | undefined>(item.vor)
+    const [overpaymentPart, setOverpaymentPart] = useState<number | undefined>(item.overpaymentPart)
+
+    useEffect(() => {
+        onUpdate(index, {
+            recipient: { tin, bic, bankAccount },
+            amount, isFeePayer, order, vor, overpaymentPart,
+        })
+    }, [amount, bankAccount, bic, index, isFeePayer, onUpdate, order, overpaymentPart, tin, vor])
+
+    return (
+        <Row align='middle' justify='space-between' gutter={GUTTER_COMPACT_ROW}>
+            <Col span={24}>
+                <Input
+                    value={tin}
+                    onChange={(e) => {
+                        setTin(e.target.value)
+                    }}
+                    prefix={`${TinTitle}:`}
+                />
+            </Col>
+            <Col span={24}>
+                <Input
+                    value={amount}
+                    onChange={(e) => {
+                        setAmount(e.target.value)
+                    }}
+                    prefix={`${AmountTitle}:`}
+                />
+            </Col>
+            <Col span={24}>
+                <Input
+                    value={order}
+                    onChange={(e) => {
+                        try {
+                            setOrder(Number(e.target.value))
+                        } catch (err) {
+                            setOrder(0)
+                        }
+                    }}
+                    prefix={`${OrderTitle}:`}
+                />
+            </Col>
+            <Col span={24}>
+                <Row gutter={0} align='middle' justify='space-between' className={styles.recipientCheckboxRow}>
+                    <Col><Typography.Text>{FeePayerTitle}:</Typography.Text></Col>
+                    <Col>
+                        <Switch size='large' checked={isFeePayer} onChange={(checked) => setIsFeePayer(checked)}/>
+                    </Col>
+                </Row>
+            </Col>
+            <Col span={24}>
+                <Row gutter={0} align='middle' justify='space-between' className={styles.recipientCheckboxRow}>
+                    <Col><Typography.Text>{VORTitle}:</Typography.Text></Col>
+                    <Col>
+                        <Switch size='large' checked={vor} onChange={(checked) => setVor(checked)}/>
+                    </Col>
+                </Row>
+            </Col>
+            <Col span={24}>
+                <Input
+                    value={overpaymentPart}
+                    onChange={(e) => {
+                        try {
+                            setOverpaymentPart(Number(e.target.value))
```

Comment:
this in theory will never throw, since `Number('something else')` is `NaN` and you'll just set `NaN` to state instead. Consider using more appropriate and explicit solution. Like:
```typescript
const value = Number(e.target.value)
setOverpaymentPart(Number.isNaN(value) ? 0 : value)
```

Reply:
Okay, refactored